### PR TITLE
PXC-3712 PXC 8.0.26 merge

### PR DESCRIPTION
**Pull request by venkatesh-prasad-v**
Testing Done
---
galera* suites: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/375/console
Failing tests: None

### CODE REVIEW

File: sql/psi_memory_key.cc [hunk 1]
```diff
@@ -405,25 +405,18 @@ static PSI_memory_info all_server_memory[] = {
...
-     PSI_DOCUMENT_ME}};
-=======
      PSI_DOCUMENT_ME},
     {&key_memory_rm_table_foreach_root, "rm_table::foreach_root",
      PSI_FLAG_THREAD, 0,
      "Mem root for temporary objects allocated while dropping tables or the "
      "whole database."},
     {&key_memory_rm_db_mdl_reqs_root, "rm_db::mdl_reqs_root", PSI_FLAG_THREAD,
+#ifdef WITH_WSREP
+     0, "Mem root for allocating MDL requests while dropping datbase."} ,
```
Comment (kamil-holubicki) [inline sql/psi_memory_key.cc]: Having 
`0, "Mem root for allocating MDL requests while dropping datbase."} ,`
part in WITH_WSREP block makes confusion.

File: sql/log_event.h [hunk 2]
```diff
@@ -42,15 +42,15 @@
...
-#include "my_aes.h"
 #include "libbinlogevents/include/binlog_event.h"
 #include "libbinlogevents/include/control_events.h"
 #include "libbinlogevents/include/load_data_events.h"
 #include "libbinlogevents/include/rows_event.h"
 #include "libbinlogevents/include/statement_events.h"
 #include "libbinlogevents/include/uuid.h"
-#include "m_string.h"     // native_strncasecmp
+#include "m_string.h"  // native_strncasecmp
+#include "m_string.h"  // native_strncasecmp
```
Comment (kamil-holubicki) [inline sql/log_event.h]: included twice

File: sql/log_event.h [hunk 3]
```diff
@@ -42,15 +42,15 @@
...
+#include "my_aes.h"
```
Comment (kamil-holubicki) [inline sql/log_event.h]: why do we need to include it in a header file?

File: libbinlogevents/include/binlog_event.h [hunk 4]
```diff
@@ -129,7 +129,8 @@
    1 + 255 /* host_len, host */ + 1U + 1 /* type, explicit_def..ts*/ + 1U +   \
    8 /* type, xid of DDL */ + 1U +                                            \
    2 /* type, default_collation_for_utf8mb4_number */ +                       \
-   1 /* sql_require_primary_key */ + 1 /* type, default_table_encryption */)
+   1 /* sql_require_primary_key */ + 1 /* type, default_table_encryption */ + \
```
Comment (kamil-holubicki) [inline libbinlogevents/include/binlog_event.h]: Is this part OK? starting from sql_require_primary_key we are storing type + value, so always 2 bytes.
Note that for variables before sql_require_primary_key there is 
`1U + xxx /* type, something */`
but starting from  sql_require_primary_key the type if not counted. Shouldnt it be like:
1 + 1 /* type, sql_require_primary_key */ \
+ 1 + 1 /* type, default_table_encryption */ + \
1 + 1 /* type, binlog_ddl_skip_rewrite */

File: sql/log_event.h (hunk 2 above)
Comment (venkatesh-prasad-v) [reply sql/log_event.h]: These changes are not introduced by this commit. Looks like they exist in Percona Server as well. https://github.com/percona/percona-server/blob/Percona-Server-8.0.26-16/sql/log_event.h#L45-L53.

File: sql/psi_memory_key.cc (hunk 1 above)
Comment (venkatesh-prasad-v) [reply sql/psi_memory_key.cc]: The idea here was to move the PXC specifinc `{&key_memory_wsrep, "wsrep", PSI_FLAG_ONLY_GLOBAL_STAT, 0, PSI_DOCUMENT_ME}};` to the end of the list.

Changed it to 
```
     {&key_memory_rm_db_mdl_reqs_root, "rm_db::mdl_reqs_root", PSI_FLAG_THREAD,
     ┆0, "Mem root for allocating MDL requests while dropping datbase."}
 #ifdef WITH_WSREP
     ,
     {&key_memory_wsrep, "wsrep", PSI_FLAG_ONLY_GLOBAL_STAT, 0, PSI_DOCUMENT_ME}
 #endif /* WITH_WSREP */
 };
```

File: libbinlogevents/include/binlog_event.h (hunk 4 above)
Comment (venkatesh-prasad-v) [reply libbinlogevents/include/binlog_event.h]: Fixed

File: sql/log_event.h (hunk 2 above)
Comment (venkatesh-prasad-v) [reply sql/log_event.h]: Fixed

**Comment by venkatesh-prasad-v** [comment]
Jenkins: https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/379/console

Failing tests:
1. sys_vars.all_vars - Fixed now.

File: sql/log_event.h (hunk 3 above)
Comment (venkatesh-prasad-v) [reply sql/log_event.h]: This change is not a part of the merge, but a side-effect of running clang-format-10 on this file.

**Review by kamil-holubicki — APPROVED** [review]
lgtm
let's wait for tests to finish before merging.

File: sql/psi_memory_key.cc [hunk 5]
```diff
@@ -411,12 +411,12 @@ static PSI_memory_info all_server_memory[] = {
      "Mem root for temporary objects allocated while dropping tables or the "
      "whole database."},
     {&key_memory_rm_db_mdl_reqs_root, "rm_db::mdl_reqs_root", PSI_FLAG_THREAD,
+     0, "Mem root for allocating MDL requests while dropping datbase."}
```
Comment (dutow) [inline sql/psi_memory_key.cc]: Typo: datbase

Comment (venkatesh-prasad-v) [reply sql/psi_memory_key.cc]: Looks a typo in upstream as well. https://github.com/mysql/mysql-server/blob/mysql-8.0.27/sql/psi_memory_key.cc#L371

**Review by dutow — APPROVED** [review]

**Comment by venkatesh-prasad-v** [comment]
> let's wait for tests to finish before merging.

@kamil-holubicki FYI the only permanent failure was `sys_vars.all_vars` and has been fixed as part of commit 0d9686b.

**Pull request merged**